packages/rulesets/src/spectral/functions/query-parameters-in-collection-get.ts

```diff
@@ -0,0 +1,38 @@
+import { getResourcesPathHierarchyBasedOnResourceType } from "./utils"
+
+export const queryParametersInCollectionGet = (pathItem: any, _opts: any, ctx: any) => {
+  if (pathItem === null || typeof pathItem !== "object") {
+    return []
+  }
+
+  const path = ctx.path || []
+  const uris = Object.keys(pathItem)
+  if (uris.length < 1) {
+    return []
+  }
+  const GET = "get"
+  const errors: any[] = []
+
+  for (const uri of uris) {
+    // skip, if GET op isn't defined
+    if (!pathItem[uri][GET]) {
+      continue
+    }
+    // check if the GET op is a collection get/list call
+    const hierarchy = getResourcesPathHierarchyBasedOnResourceType(uri)
+    if (hierarchy.length == 0 && pathItem[uri][GET]) {
```

Comment:
I think this is not needed since it's checked on line 18 already
```suggestion
    if (hierarchy.length == 0) {
```